syslogish/server.go

```diff
@@ -106,13 +106,9 @@ func (s *Server) receive() {
 		if err != nil {
 			log.Fatal("syslogish server read error", err)
 		}
-		rawMessage := strings.TrimSuffix(string(buf[:n]), "\n")
-		// We want to strip off all the leading syslog junk but there is no good way to do that.
-		// So we split the string on the leading "{" but we have to add it back so we can parse the log message into a map
-		logMessage := "{" + strings.SplitAfterN(rawMessage, "{", 2)[1]
-
+		message := strings.TrimSuffix(string(buf[:n]), "\n")
 		select {
-		case s.storageQueue <- logMessage:
+		case s.storageQueue <- message:
```

Comment:
I'm honestly not sure.  The _original_ intent of this code was to dump messages into a channel as fast as possible to avoid the UDP stack's own internal queue filling up and dropping _new_ incoming messages.  So if and when the queue backs up, the scheme basically favors _new_ messages that we might still be able to process over the one we have in hand right now.

Are there better ways to deal with this?  Probably.  But since the behavior isn't really changed by the PR, that should probably be a separate issue if it's worthy of further investigation.